src/core/qgsvectorlayer.cpp

```diff
@@ -2362,8 +2339,33 @@ bool QgsVectorLayer::readSymbology( const QDomNode &layerNode, QString &errorMes
     mGeometryOptions->readXml( layerNode.namedItem( QStringLiteral( "geometryOptions" ) ) );
 
   if ( categories.testFlag( Forms ) )
+  {
     mEditFormConfig.readXml( layerNode, context );
 
+    // Load editor widget configuration
+    QDomElement widgetsElem = layerNode.namedItem( QStringLiteral( "fieldConfiguration" ) ).toElement();
+    QDomNodeList fieldConfigurationElementList = widgetsElem.elementsByTagName( QStringLiteral( "field" ) );
+    for ( int i = 0; i < fieldConfigurationElementList.size(); ++i )
+    {
+      const QDomElement fieldConfigElement = fieldConfigurationElementList.at( i ).toElement();
+      const QDomElement fieldWidgetElement = fieldConfigElement.elementsByTagName( QStringLiteral( "editWidget" ) ).at( 0 ).toElement();
+
+      QString fieldName = fieldConfigElement.attribute( QStringLiteral( "name" ) );
+      mFieldConfigurationFlags[fieldName] = qgsFlagKeysToValue( fieldConfigElement.attribute( QStringLiteral( "configurationFlags" ) ), QgsField::ConfigurationFlag::None );
```

Comment:
this is not part of the form category but part of the fields.

Reply:
@3nids You configure this in *Forms* properties tab, but when user wants to copy/paste this configuration he has to select *Fields* (or now *Field and Widget configuration* since it has been renamed in  #38786) and not *Forms*. I find it a little bit confusing, like said in the issue  #37702. 

Am I missing something here?

Reply:
The configuration flags are defined in the fields tab.

The widgets config is also part of "fields and widget config" as rewritten in https://github.com/qgis/QGIS/pull/38786/files

Forms is only about the layout/UI.

So, I think you moved this to the wrong section.

Reply:
I agree for the configuration flags, it should be in fields part.

Still, for the widget configuration, my proposal was to move them in Forms so when user copy/paste *Forms* it's exactly what you configure in *Forms* properties tab. No more, no less. 

Wouldn't be better than rename the items?

Reply:
No strong opinion
So you would revert https://github.com/qgis/QGIS/pull/38786 ?

Reply:
If no one is against, yes

Reply:
ping @nyalldawson

Reply:
No objection here